File: composeApp/src/desktopMain/kotlin/fr/thomasbernard03/androidtools/domain/usecases/file/DeleteFileUseCase.kt
package fr.thomasbernard03.androidtools.domain.usecases.file

import fr.thomasbernard03.androidtools.domain.repositories.FileRepository
import org.koin.java.KoinJavaComponent.get

class DeleteFileUseCase(
    private val fileRepository: FileRepository = get(FileRepository::class.java)
) {
    suspend operator fun invoke(path: String) {
        fileRepository.deleteFile(path)
    }
}

File: lib/features/file_explorer/domain/usecases/delete_file_usecase.dart
import 'package:android_tools/features/file_explorer/domain/repositories/file_repository.dart';

class DeleteFileUsecase {
  final FileRepository _fileRepository;

  DeleteFileUsecase(this._fileRepository);

  Future<void> call(String filePath, String deviceId) {
    return _fileRepository.deleteFile(filePath, deviceId);
  }
}
